tests/ui/consts/const-fn-error.stderr

```diff
@@ -1,37 +1,20 @@
-error[E0658]: `for` is not allowed in a `const fn`
-  --> $DIR/const-fn-error.rs:5:5
-   |
-LL | /     for i in 0..x {
-LL | |
-LL | |
-LL | |
-LL | |         sum += i;
-LL | |     }
-   | |_____^
-   |
-   = note: see issue #87575 <https://github.com/rust-lang/rust/issues/87575> for more information
-   = help: add `#![feature(const_for)]` to the crate attributes to enable
```

Comment:
This helpful hint seems to have been lost?

Reply:
Notably, enabling this gate is neither necessary nor sufficient under const traits. I should have noted that.

If and when we work out how cons stability works for const traits it may, but it's just misleading today.

Reply:
It's just quite frustrating to get a nightly feature error without any hint about what the feature gate is. If I first get an error saying to enable const_for, and then another error saying to enable const_trait_impl, that's a lot better IMO.

Or does the const_for gate not actually do anything right now?

Reply:
AFAICT it does not do anything.

Reply:
Ah. Then it should mention `const_trait_impl`, no?

Reply:
No. Iterator is not a const trait yet.

Reply:
Oh, I see.

Yeah then the new error is fine. Maybe entirely remove the `const_for` symbol (and the others that were used only in the HIR const checker).

Reply:
Well we can't remove the symbol bc those are still used for the lang feature declaration. Even if we were to deprecate the lang feature, it would need to go in the "removed" table which still needs a symbol afaict.

Reply:
Ah yeah the lang feature(s) should be removed then, possibly in a future PR. Bummer that we have to keep the symbol.

Or do we imagine using this lang feature again in the future? The way things are going now, it looks more like there will be libs features for the IntoIterator and Iterator traits, and those will be required for `for` loops.

Reply:
Yeah, I don't believe we'll need the features again, but I'll remove them in a future PR. All of these should desugar to const library features.